### PR TITLE
gcp - Cloud Storage initial setup with nested resources

### PR DESCRIPTION
**Pull request by Dkhanas**
Added initial classes with query and get for BucketAccessControls, DefaultObjectAccessControls, Objects , ObjectAccessControls and Notifications data resource in Cloud Storage service

### CODE REVIEW

**Comment by Dkhanas** [comment]
@jondavidi  I created new PR rebase from upstream master branch, now it is up to date. It related to closed PR https://github.com/mediapills/cloud-custodian/pull/41

File: tools/c7n_gcp/c7n_gcp/resources/storage.py [hunk 1]
```diff
@@ -25,8 +30,140 @@ class resource_type(TypeInfo):
...
+        }
+
+        @staticmethod
+        def get(client, resource_info):
+            info = client.execute_command(
+                'get', {
+                    'bucket': resource_info['bucket_name'],
+                    'entity': resource_info['entity']
+                })
+            info['bucket_name'] = resource_info['bucket_name']
```
Comment (jondavidi) [inline tools/c7n_gcp/c7n_gcp/resources/storage.py]: why is this required to change the bucket name in the response?

File: tools/c7n_gcp/tests/test_gcp_storage.py [hunk 2]
```diff
@@ -31,15 +31,158 @@ def test_bucket_query(self):
...
+        factory = self.replay_flight_data('bucket-default-object-access-control-query', project_id)
+        p = self.load_policy(
+            {'name': 'all-bucket-default-object-access-control',
+             'resource': 'gcp.bucket-default-object-access-control'},
+            session_factory=factory)
+        resources = p.run()
+        self.assertEqual(len(resources), 1)
+        self.assertEqual(resources[0]['entity'], entity)
+
+    def test_bucket_get(self):
```
Comment (jondavidi) [inline tools/c7n_gcp/tests/test_gcp_storage.py]: this test doesn't seem to be using the event data

File: tools/c7n_gcp/tests/test_gcp_storage.py [hunk 3]
```diff
@@ -31,15 +31,158 @@ def test_bucket_query(self):
...
+        factory = self.replay_flight_data('bucket-object-query', project_id)
+        p = self.load_policy(
+            {'name': 'all-bucket-object',
+             'resource': 'gcp.bucket-object'},
+            session_factory=factory)
+        resources = p.run()
+        self.assertEqual(len(resources), 1)
+        self.assertEqual(resources[0]['name'], "commit-example.txt")
+
+    def test_bucket_get(self):
```
Comment (jondavidi) [inline tools/c7n_gcp/tests/test_gcp_storage.py]: this test doesn't seem to be using event data

File: tools/c7n_gcp/tests/test_gcp_storage.py [hunk 4]
```diff
@@ -31,15 +31,158 @@ def test_bucket_query(self):
...
+        factory = self.replay_flight_data('bucket-object-access-control-query', project_id)
+        p = self.load_policy(
+            {'name': 'all-bucket-object-access-control',
+             'resource': 'gcp.bucket-object-access-control'},
+            session_factory=factory)
+        resources = p.run()
+        self.assertEqual(len(resources), 1)
+        self.assertEqual(resources[0]['object'], "commit-example.txt")
+
+    def test_bucket_get(self):
```
Comment (jondavidi) [inline tools/c7n_gcp/tests/test_gcp_storage.py]: this test doesn't seem to be using event data

Comment (Dkhanas) [reply tools/c7n_gcp/tests/test_gcp_storage.py]: Updated

File: tools/c7n_gcp/tests/test_gcp_storage.py (hunk 3 above)
Comment (Dkhanas) [reply tools/c7n_gcp/tests/test_gcp_storage.py]: Updated

File: tools/c7n_gcp/tests/test_gcp_storage.py (hunk 2 above)
Comment (Dkhanas) [reply tools/c7n_gcp/tests/test_gcp_storage.py]: Updated

File: tools/c7n_gcp/c7n_gcp/resources/storage.py (hunk 1 above)
Comment (Dkhanas) [reply tools/c7n_gcp/c7n_gcp/resources/storage.py]: I don't change bucket name. I add it in child resource info because by default resource don't have bucket name in response. It added because parent resource require bucket name.

**Comment by jondavidi** [comment]
@Dkhanas this pull request commit history is messed up like another we had discussed previously. Please fix this and please see if there's a way to avoid this in the future, I believe it's somethign to do with how you're merging in new changes and pushing

**Comment by Dkhanas** [comment]
@jondavidi for avoid this on the future, we can create git hook. But for now we do not have enough permission

**Comment by mediapills** [comment]
merged into #119
